api/app_analytics/cache.py

```diff
@@ -31,3 +33,51 @@ def track_request(self, resource: int, host: str, environment_key: str):
             self._cache[key] += 1
         if (timezone.now() - self._last_flushed_at).seconds > CACHE_FLUSH_INTERVAL:
             self._flush()
+
+
+class FeatureEvaluationCache:
+    def __init__(self):
+        self._cache = {}
+        self._last_flushed_at = timezone.now()
+
+    def _flush(self):
+        evaluation_data = {}
+        for (environment_id, feature_name), eval_count in self._cache.items():
+            if environment_id in evaluation_data:
+                evaluation_data[environment_id][feature_name] = eval_count
+            else:
+                evaluation_data[environment_id] = {feature_name: eval_count}
+
+        for environment_id, feature_evaluations in evaluation_data.items():
+            if settings.USE_POSTGRES_FOR_ANALYTICS:
+                track_feature_evaluation.delay(
+                    kwargs={
+                        "environment_id": environment_id,
+                        "feature_evaluations": feature_evaluations,
+                    }
+                )
+
+            elif settings.INFLUXDB_TOKEN:
+                track_feature_evaluation_influxdb.delay(
+                    kwargs={
+                        "environment_id": environment_id,
+                        "feature_evaluations": feature_evaluations,
+                    }
+                )
```

Comment:
nit: I feel that the postgres/influx discrimination belongs inside the task itself. Moreover, this should improve analytics backend migration for self-hosting customers.

Reply:
Yeah, I agree with you, but that's a different pull request I think(because there are two v2 tasks as well)